src/audio/base_fw.c

```diff
@@ -252,6 +263,15 @@ static uint32_t basefw_set_system_time(uint32_t param_id,
 	global_system_time_info.dsp_time.val_l = (uint32_t)(current_dsp_time);
 	global_system_time_info.dsp_time.val_u = (uint32_t)(current_dsp_time >> 32);
 
+	/* use default timestamp if 64bit is not enabled since 64bit is necessary for host time */
+	if (!IS_ENABLED(CONFIG_LOG_TIMESTAMP_64BIT))
+		return IPC4_SUCCESS;
+
+	host_time = k_ns_to_cyc_ceil64(*(uint64_t *)data + TIME_SETTING_LATENCY_NS);
+	time_delta = host_time - current_dsp_time;
+	log_set_timestamp_func(basefw_get_timestamp,
+			       sys_clock_hw_cycles_per_sec());
```

Comment:
What happens when the DSP frequency changes?

Also, what happens on IMR resume?

Reply:
The time is queried from wall clock hardware, not based on DSP frequency, so no problem. For IMR resume case, host will set timestamp again and we will sync time again.

Reply:
38.4mhz clock